vars/cloudFoundryDeploy.groovy

```diff
@@ -290,17 +314,41 @@ private void handleCFNativeDeployment(Map config, script) {
     echo "[${STEP_NAME}] - cfManifest=${config.cloudFoundry.manifest}"
     echo "[${STEP_NAME}] - cfManifestVariables=${config.cloudFoundry.manifestVariables?:'none specified'}"
     echo "[${STEP_NAME}] - cfManifestVariablesFiles=${config.cloudFoundry.manifestVariablesFiles?:'none specified'}"
+    echo "[${STEP_NAME}] - cfdeployDockerImage=${config.cloudFoundry.deployDockerImage ?: 'none specified'}"
+    echo "[${STEP_NAME}] - cfdockerCredentialsId=${config.cloudFoundry.dockerCredentialsId ?: 'none specified'}"
     echo "[${STEP_NAME}] - smokeTestScript=${config.smokeTestScript}"
 
     checkIfAppNameIsAvailable(config)
-    dockerExecute(
-        script: script,
-        dockerImage: config.dockerImage,
-        dockerWorkspace: config.dockerWorkspace,
-        stashContent: config.stashContent,
-        dockerEnvVars: [CF_HOME: "${config.dockerWorkspace}", CF_PLUGIN_HOME: "${config.dockerWorkspace}", STATUS_CODE: "${config.smokeTestStatusCode}"]
-    ) {
-        deployCfNative(config)
+
+    def dockerCredentials = []
+    if (config.cloudFoundry.dockerCredentialsId != null && config.cloudFoundry.dockerCredentialsId != '') {
+        dockerCredentials.add(usernamePassword(
+            credentialsId: config.cloudFoundry.dockerCredentialsId,
+            passwordVariable: 'dockerPassword',
+            usernameVariable: 'dockerUsername'
+        ))
+    }
+
+    withCredentials([usernamePassword(
+        credentialsId: config.cloudFoundry.credentialsId,
+        passwordVariable: 'password',
+        usernameVariable: 'username'
+    )] + dockerCredentials) {
+        dockerExecute(
+            script: script,
+            dockerImage: config.dockerImage,
+            dockerWorkspace: config.dockerWorkspace,
+            stashContent: config.stashContent,
+            dockerEnvVars: [
+                CF_HOME           : "${config.dockerWorkspace}",
+                CF_PLUGIN_HOME    : "${config.dockerWorkspace}",
+                // if the Docker registry requires authentication the DOCKER_PASSWORD env variable must be set
+                CF_DOCKER_PASSWORD: "${binding.hasVariable("dockerPassword") ? dockerPassword : ''}",
```

Comment:
CF_DOCKER_PASSWORD: "${dockerPassword  ?: ''}"

Reply:
You could try `CF_DOCKER_PASSWORD: "${env.dockerPassword ?: ''}"`

Reply:
withCrednetials should also set an environment variable, if you use the name `CF_DOCKER_PASSWORD` in line 318 you do not have to handle it here.
Same might be true for the username if you also can inject the username via env variables.